indico/modules/rb_new/client/js/common/timeline/DailyTimelineContent.jsx

```diff
@@ -70,33 +72,44 @@ export default class DailyTimelineContent extends React.Component {
         return onClickLabel ? () => onClickLabel(id) : false;
     };
 
+    hasUsage = ({preBookings, blockings, bookings, nonbookablePeriods, unbookableHours}) => {
+        return !(_.isEmpty(preBookings) && _.isEmpty(blockings) && _.isEmpty(nonbookablePeriods) &&
+                 _.isEmpty(bookings) && _.isEmpty(nonbookablePeriods) && _.isEmpty(unbookableHours));
+    };
+
     renderTimelineRow = ({availability, label, conflictIndicator, key, room}) => {
         const {
-            minHour, maxHour, hourStep, itemClass: ItemClass, itemProps, recurrenceType, onClick, longLabel
+            minHour, maxHour, hourStep, itemClass: ItemClass, itemProps, recurrenceType, onClick, longLabel, showUnused
         } = this.props;
         const columns = ((maxHour - minHour) / hourStep) + 1;
+
         // TODO: Consider plan B (availability='alternatives') option when implemented
         const hasConflicts = !(_.isEmpty(availability.conflicts) && _.isEmpty(availability.preConflicts));
-        return (
-            <div styleName="timeline-row" key={`element-${key}`}>
-                <TimelineRowLabel label={label}
-                                  availability={conflictIndicator ? (hasConflicts ? 'conflict' : 'available') : null}
-                                  longLabel={longLabel}
-                                  onClickLabel={this.onClickLabel(room.id)} />
-                <div styleName="timeline-row-content" style={{flex: columns}}>
-                    <ItemClass startHour={minHour} endHour={maxHour} data={availability} room={room}
-                               onClick={() => {
-                                   if (onClick && (!hasConflicts || recurrenceType !== 'single')) {
-                                       onClick(room);
-                                   }
-                               }}
-                               setSelectable={selectable => {
-                                   this.setState({selectable});
-                               }}
-                               {...itemProps} />
+
+        if (!showUnused && !this.hasUsage(availability)) {
+            return null;
+        } else {
```

Comment:
no need for an else block here. removing it also removes the extra level of indentation on the code below